azure-quantum/tests/quantum_test_base.py

```diff
@@ -0,0 +1,197 @@
+    @property
+    def tenant_id(self):
+        return self._tenant_id
+
+    @property
+    def resource_group(self):
+        return self._resource_group
+
+    @property
+    def subscription_id(self):
+        return self._subscription_id
+
+    @property
+    def workspace_name(self):
+        return self._workspace_name
+
+    def create_workspace(self) -> Workspace:
+        """Create workspace using credentials stored in config file
+
+        :return: Workspace
+        :rtype: Workspace
+        """
+
+        workspace = Workspace(
+        
+            subscription_id=self.subscription_id,
+            resource_group=self.resource_group,
+            name=self.workspace_name,
+        )
+
+        if self.is_live or self.in_recording:
+            workspace.credentials = ServicePrincipalCredentials(
+                tenant=self.tenant_id,
+                client_id=self.client_id,
+                secret=self.client_secret,
+                resource  = "https://quantum.microsoft.com"
+            )
+            workspace.login(False)
+        else:
+            workspace.credentials = BasicTokenAuthentication(token=self.dummy_auth_token)
+
+        return workspace
+
+class CustomRecordingProcessor(RecordingProcessor):
+
+    ALLOW_HEADERS = [
+        'content-length',
+        'content-type',
+        'accept',
+        'accept-encoding',
+        'accept-charset',
+        'accept-ranges',
+        'x-ms-range',
+        'transfer-encoding',
+        'x-ms-blob-content-md5',
+        'x-ms-blob-type',
+        'x-ms-creation-time',
+        'x-ms-lease-state',
+        'x-ms-lease-status',
+        'x-ms-server-encrypted',
+        'x-ms-version'
+        ]
+    
+    def __init__(self):
+        self._regexes = []
+
+    def register_regex(self, oldRegex, new):
+        self._regexes.append((re.compile(oldRegex), new))
+
+    def process_request(self, request):
+        headers = {}
+        for key in request.headers:
+            if key.lower() in self.ALLOW_HEADERS:
+                headers[key] = request.headers[key]
+        #request.headers = headers
+
+        for oldRegex, new in self._regexes:
+            request.uri = oldRegex.sub(new, request.uri)            
+
+        if _get_content_type(request) == "application/x-www-form-urlencoded":
+            body = request.body.decode("utf-8")
+            for oldRegex, new in self._regexes:
+                body = oldRegex.sub(new, body)
+            request.body = body.encode("utf-8")
+        else:
+            body = str(request.body)
+            for oldRegex, new in self._regexes:
+                body = oldRegex.sub(new, body)
+            request.body = body
+
+        return request
+
+    def process_response(self, response):
+        import six
+
+        headers = {}
+        for key in response['headers']:
+            if key.lower() in self.ALLOW_HEADERS:
+                headers[key.lower()] = response['headers'][key]
+        #response['headers'] = headers
```

Comment:
```suggestion
```

Reply:
I removed the comment for this line to re-enable filtering which headers we want in the recordings with PR #68.